symengine/tests/basic/test_sets.cpp

```diff
@@ -690,16 +698,22 @@ TEST_CASE("ConditionSet : Basic", "[basic]")
     REQUIRE(eq(*r1->contains(integer(3)), *boolTrue));
     REQUIRE(eq(*r1->contains(integer(-3)), *boolFalse));
     REQUIRE(eq(*r1->contains(integer(2)), *boolFalse));
-    REQUIRE(eq(*r1->contains(finiteset({integer(3)})), *boolTrue));
+    REQUIRE(eq(*r1->contains(finiteset({integer(3)})), *boolFalse));
```

Comment:
I think that the problem is with `Interval:: contains`
```
diff --git a/symengine/logic.cpp b/symengine/logic.cpp
index 979ff74..24820e2 100644
--- a/symengine/logic.cpp
+++ b/symengine/logic.cpp
 RCP<const Boolean> contains(const RCP<const Basic> &expr,
                             const RCP<const Set> &set)
 {
-    if (is_a_Number(*expr)) {
-        return set->contains(expr);
-    } else {
-        return make_rcp<Contains>(expr, set);
-    }
+    return set->contains(expr);
 }
 
diff --git a/symengine/sets.cpp b/symengine/sets.cpp
--- a/symengine/sets.cpp
+++ b/symengine/sets.cpp
 
 RCP<const Boolean> Interval::contains(const RCP<const Basic> &a) const
 {
-    if (not is_a_Number(*a))
-        return make_rcp<Contains>(a, rcp_from_this_cast<const Set>());
+    if (not is_a_Number(*a)) {
+        if (is_a<Symbol>(*a) or is_a<Constant>(*a))
+            return make_rcp<Contains>(a, rcp_from_this_cast<const Set>());
+        return boolean(false);
+    }

```
^Above patch should fix this problem. thoughts @isuruf and @nishnik?